README.md

```diff
@@ -1,41 +1,108 @@
+#### iOS
 
-## Getting Started
+Add the NSLocationWhenInUsageDescription and NSLocationUsageDescription values to your info.plist file. This is the message that is displayed to your user when the permission for their location is requested. For instance,
+
+```
+<key>NSLocationWhenInUsageDescription</key>
+<string>TODO NSLocationWhenInUsageDescription</string>
+<key>NSLocationUsageDescription</key>
+<string>TODO NSLocationUsageDescription</string>
+```
+
+#### Android
+
+Add the location permission to your AssemblyInfo.cs file.
+
+```
+[assembly: UsesPermission(Android.Manifest.Permission.AccessFineLocation)]
+```
+
+### Request permission
+
+After you have added the relevant location permission info to your project configuration, locate the spot in your code where you want to ask the user for their location permission. This must be done before accessing their location. Add the following line to do so:
+
+```
+var isGranted = await locationService.RequestPermission(ct);
+```
+
+You may safely request the permission multiple times; the user will only get prompted as often as is necessary.
+
+For instance, if you ask the user for their permission once and they decline, calling ```RequestPermission``` again will not prompt the user again, but simply return ```false```.
+
+If the user answers "Allow once", then calling ```RequestPermission``` again during the lifetime of the app will simply return ```true```. Once the app is restarted, calling ```RequestPermission``` will prompt them for the permission once again, since their original permission has now expired.
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+Once permission is requested, you can use the various methods and extension methods of IGeolocatorService.
 
 ## Features
 
-{More details/listing of features of the project}
+### Request the user's location
+
+Once you have obtained the location permission, it's a simple matter to obtain the location:
+
+```
+var location = await locationService.GetLocation(ct);
+```
+
+If the user has denied the location permission, an ```InvalidOperationException``` is thrown; therefore, we recommend not calling this method if you know the user has denied the location permission and this is a normal scenario. If the app has not kept track of this information, we recommend calling this extension method instead:
+
+```
+var locationOrNull = await locationService.GetLocationOrNull(ct);
+```
+
+This method will return null instead of throwing an exception, in case the permission was denied.
+
+### React to changes in the user's location or their permissions
+
+```IGeolocatorService``` offers events to allow you to react when the user's location has changed, or when they have granted or denied permissions. Even if these changes occur while the app is in background, they will be raised when the app comes back to the foreground.
+
+```
+locationService.LocationChanged -= RefreshUserLocation;
```

Comment:
I think I would write OnLocationChanged just for consistency :)

Reply:
Not sure if we want to have the -= in the documentation. We often do it but it depends on the scenario, if it's in a constructor there's no point.